COMMANDS.org

```diff
@@ -0,0 +1,117 @@
+
+#+begin_quote
+/Question: I'm used to tools like brew - how do I use nix in a similar way?/
+#+end_quote
+
+If your workflow has involved...
+- ~brew install some_package~
+- ~port install some_package~
+- ~api-get install some_package~
+
+...and are making the move to ~nix~, then you are in the right place! ~nix~ can be used as a drop in replacement for these tools on most linux/unix based systems.
+
+*** Install a package
+
+Once the desired package has been identified, use ~nix-env~ to install the package globally...
+
+#+begin_src bash
+  nix-env -i some_package
+#+end_src
+
+- ~-i~: short flag for ~--install~.
+  - Takes multiple arguements in a space-seperated list.
+  - By default, ~nix-env -i~ will do a lookup of the argument given against ~nixpkgs~, similarly to going to https://search.nixos.org/, searching, and using the top result.
+- ~-A~: short for ~--attr~ across many ~nix~ commands.
+  - Instead of relying on the lookup, which could be slow at times, one could add ~-A~ along with the /exact/ package name to skip the lookup completely. This requires the name given to be exact, and can allow for installing a specific version of the target package.
+  - Can be combined with other flags i.e. ~-iA~
+
+#+begin_quote
+Note: If a package being installed complains about the package license being "unfree", this means the package has some licensing agreement that is not conguent with typical open-source tools.
+
+To install these packages anyway, prepend ~NIXPKGS_ALLOW_UNFREE=1~ to the call to make a 1 time allowance. The whole call should look something like:
+#+begin_src bash
+  NIXPKGS_ALLOW_UNFREE=1 nix-env -i some_package
+#+end_src
+#+end_quote
+
+*** Uninstalling a package
+
+Uninstalling a package requires knowing its name. We can find a list of the installed packages like this:
+
+#+begin_src bash
+  nix-env -q some_package
+#+end_src
+
+With the target package name in hand, we can then run:
+
+#+begin_src bash
+  nix-env -e some_package
+#+end_src
+
+- ~-e~: Stands for "erase"
+  - Takes a space-seperated list of pacakges to be removed from the system
+
+To validate, simply re-run ~nix-env -q some_package~ and check that output.
+
+** 📝 Declaritive Pacakge Management
+
+To be detailed later.
+
+There are a few ways to approach this, including using ~flakes~ or helper packages installed imperatively like ~home-manager~
+
+** 🗑️ Temporary Packages
+
+Want to test a new tool or utility without making any permanent changes to your system? Lets test this with a couple of system info tools.
+
+Using ~nix-shell~, we can drop into a new temporary shell, adding any number of packages to this temporary shell with the ~-p~ flag 
+#+begin_src bash
+  nix-shell -p neofetch
+#+end_src
+
+That's it! You can add as many packages as you like to the ~-p~ flag, seperateing each package with a single space. Upon ~exit~ of the shell, all the "installed" packages are removed from your envrionment.
+
+*** Validate
+
+Once in the temporary shell:
```

Comment:
Mention something about trying it with `neofetch`, and also link to the documentation for `neofetch`

Reply:
Not clear here - the entire "Temporary" section is about using neofetch already

Added more ~neofetch~ to things

Reply:
More of what I mean is link to neofetch so the person knows what to expect with the example package